File: go/module-config/push-to-function/fs/bar-bf.go
/*
 *  Copyright © 2021 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the Apache 2.0 license
 *  that can be found in the LICENSE file.
 */

package fs

import (
	"fmt"
)

type BarBfCfgInfo struct {
	Z int
}

type BarBfCfgSrc struct {
	Get func() BarBfCfgInfo
}

type BarBfT = func()

func BarBfC(cfg BarBfCfgSrc) BarBfT {
	return func() {
		fmt.Println(cfg.Get().Z)
	}
}

File: kt/main/kotlin/moduleconfig/pushtofunction/fs/barBf.kt
/*
 *  Copyright © 2021 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the Apache 2.0 license
 *  that can be found in the LICENSE file.
 */

package tryout.moduleconfig.pushtofunction.fs

data class BarBfCfgInfo(
	val z: Int
)

data class BarBfCfgSrc(
	val get: () -> BarBfCfgInfo
)

typealias BarBfT = () -> Unit

fun barBfC(cfg: BarBfCfgSrc): BarBfT {
	return { println(cfg.get().z) }
}
